File: CPP Python/Coprime.py
#Problem:https://codeforces.com/problemset/problem/1742/D
#Name:Coprime
#Author:Vedant Borkar
from math import gcd
def coprime(n,max):
  l=[int(x) for x in input().split()]
  for i in range(n):
    for j in range(n):
      if(gcd(l[i],l[j])==1):
        if((i+j)>max):
          max=i+j
  if(max>-1):print(max+2)  
  else: print("-1")
t=int(input())
while(t>0):
  coprime(int(input()),-1)
  t-=1

File: Coprime.cpp
//Problem:https://codeforces.com/problemset/problem/1742/D
//Name: Coprime
//Author:Vedant Borkar
#include<bits/stdc++.h>
using namespace std;
int coprime(int n, int max) {
    vector<int> l(n);
    for(int i=0; i<n; i++)
        cin >> l[i];
    for(int i=0; i<n; i++) {
        for(int j=0; j<n; j++) {
            if(__gcd(l[i],l[j])==1) {
                if((i+j)>max)
                    max=i+j;
            }
        }
    }
    if(max>-1)
        cout << max+2 << "\n";
    else
        cout << "-1" << "\n";
    return 0;
}
int main() {
    int t,n;
    cin >> t;
    while(t--) {cin>>n;
        coprime(n,-1);
    }
    return 0;
}
